core/src/main/java/org/mskcc/cbio/oncokb/cache/CacheFetcher.java

```diff
@@ -244,6 +249,42 @@ public Alteration getAlterationFromGenomeNexus(GNVariantAnnotationType gnVariant
         return AlterationUtils.getAlterationFromGenomeNexus(gnVariantAnnotationType, genomicLocation, referenceGenome);
     }
 
+    public void cacheAlterationFromGenomeNexus(GenomeNexusAnnotatedVariantInfo gnAnnotatedVariantInfo) {
+        Alteration alteration = new Alteration();
+
+        // Build the Alteration object from the pre-annotated GN variant object.
+        if (StringUtils.isNotEmpty(gnAnnotatedVariantInfo.getHugoSymbol())) {
+            Gene gene = GeneUtils.getGene(gnAnnotatedVariantInfo.getHugoSymbol());
+            if (gene == null) {
+                gene = new Gene();
+                gene.setHugoSymbol(gnAnnotatedVariantInfo.getHugoSymbol());
+                gene.setEntrezGeneId(gnAnnotatedVariantInfo.getEntrezGeneId());
+            }
+            alteration.setGene(gene);
+        }
+        alteration.setAlteration(gnAnnotatedVariantInfo.getHgvspShort());
+        alteration.setProteinStart(gnAnnotatedVariantInfo.getProteinStart());
+        alteration.setProteinEnd(gnAnnotatedVariantInfo.getProteinEnd());
+        alteration.setConsequence(VariantConsequenceUtils.findVariantConsequenceByTerm(gnAnnotatedVariantInfo.getConsequenceTerms()));
+
+        String hgvsg = gnAnnotatedVariantInfo.getHgvsg();
+        String genomicLocation = gnAnnotatedVariantInfo.getGenomicLocation();
+        ReferenceGenome referenceGenome = gnAnnotatedVariantInfo.getReferenceGenome();
+
+        // Store pre-annotated alteration into Redis cache
+        Cache cache = cacheManager.getCache(CacheCategory.GENERAL.getKey() + REDIS_KEY_SEPARATOR + "getAlterationFromGenomeNexus");
+        if (StringUtils.isNotEmpty(hgvsg)) {
+            String cacheKey = org.springframework.util.StringUtils.arrayToDelimitedString(new Object[]{GNVariantAnnotationType.HGVS_G, referenceGenome, hgvsg}, REDIS_KEY_SEPARATOR);
```

Comment:
I would specify the content being String. `new String[]{A, B}`, then do a String.join();